### PR TITLE
Append server name from connection string to error message

### PR DESCRIPTION
**Pull request by lilgreenbird**
if server name in connection string is different from connect server then append server name to error message thrown

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java [hunk 1]
```diff
@@ -163,6 +163,25 @@ SharedTimer getSharedTimer() throws SQLServerException {
...
+    /**
+     * Get the server name string including redirected server if applicable
+     * 
+     * @param serverName
+     * @return
+     */
+    String getServerNameString(String serverName) {
+        String serverNameFromConnectionStr = activeConnectionProperties
+                .getProperty(SQLServerDriverStringProperty.SERVER_NAME.toString());
+        if (serverName.equals(serverNameFromConnectionStr)) {
```
Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java:175]: do null check on serverName

**Comment by peterbae** [comment]
Can you add tests to this (if it's feasible) so that the test shows what the error message would look like instead?

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java [hunk 2]
```diff
@@ -163,6 +163,25 @@ SharedTimer getSharedTimer() throws SQLServerException {
...
     }
 
+    /**
+     * Get the server name string including redirected server if applicable
+     * 
+     * @param serverName
+     * @return
+     */
+    String getServerNameString(String serverName) {
+        String serverNameFromConnectionStr = activeConnectionProperties
```
[the remark this answers is not in the document]
Comment (ulvii) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java:173]: this is not resolved

**Comment by lilgreenbird** [comment]
[mssql-jdbc-8.3.1-SNAPSHOT.jre14-preview.zip](https://github.com/microsoft/mssql-jdbc/files/4549094/mssql-jdbc-8.3.1-SNAPSHOT.jre14-preview.zip)
[mssql-jdbc-8.3.1-SNAPSHOT.jre11-preview.zip](https://github.com/microsoft/mssql-jdbc/files/4549095/mssql-jdbc-8.3.1-SNAPSHOT.jre11-preview.zip)
[mssql-jdbc-8.3.1-SNAPSHOT.jre8-preview.zip](https://github.com/microsoft/mssql-jdbc/files/4549096/mssql-jdbc-8.3.1-SNAPSHOT.jre8-preview.zip)

**Pull request merged**